numberOfProvinces.js

```diff
@@ -0,0 +1,106 @@
+/*
+n cities
+
+a - b - c
+
+i: n x n matrix
+o: total number of provinces
+
+directly or indirectly
+
+Input: isConnected = [[1,1,0],[1,1,0],[0,0,1]]
+Output: 2
+
+   0 1 2
+0 [1,1,0],
+1 [1,1,0],
+2 [0,0,1]
+
+*/
+
+const { Queue } = require('../utils/queue');
+
+const directions = [
+    [1, 0],
+    [0, 1],
+    [-1, 0],
+    [0, -1],
+];
+
+const CONNECTED = 1;
+
+const getRowColString = (row, col) => `${row}, ${col}`;
+
+const isBound = (row, col, matrix) => {
+    const rowBound = matrix.length;
+    const colBound = matrix[0].length;
+
+    const isRowBound = row >= 0 && row < rowBound;
+    const isColBound = col >= 0 && col < colBound;
+
+    return isRowBound && isColBound;
+};
+
+const getChildren = (row, col, matrix) => {
+    const children = [];
+
+    for (const direction of directions) {
+        const [rowDir, colDir] = direction;
+
+        const newRow = row + rowDir;
+        const newCol = col + colDir;
+
+        const isInBound = isBound(newRow, newCol, matrix);
+        if (!isInBound) continue;
+        if (matrix[newRow][newCol] !== CONNECTED) continue;
+
+        children.push([row, col]);
+    }
+
+    return children;
+};
+
+const markedVisited = (row, col, visited, matrix) => {
```

Comment:
What are you marking as visited? Include that in the name. You could say something like `markProvinceAsVisited`. Alterantively, if you want to make this function generic, you can name it something like `markComponentAsVisited` (I personally think the latter is better).